### ISSUE TITLE
`panic: runtime error: invalid memory address or nil pointer dereference` with `nyx clean`

### ISSUE THREAD

**Issue by rrzal** [mooltiverse/nyx#217]
`.gitlab-ci.yml`:
```yml
infer-job:
  stage: infer
  tags:
    - shell
  script:
    - git checkout -b $CI_COMMIT_BRANCH
    - nyx clean
    - nyx
    - python3 ./ci_scripts/parse_nyx_infer_result.py
  artifacts:
    reports:
      dotenv: build.env
```

log:
```console
Running with gitlab-runner 15.11.0 (436955cb)
  on xxx
Preparing the "shell" executor 00:00
Using Shell (bash) executor...
Preparing environment 00:00
Running on ci-station...
Getting source from Git repository 00:02
Fetching changes...
Initialized empty Git repository in /home/gitlab-runner/builds/xxx/.git/
Created fresh repository.
Checking out 54477a96 as detached HEAD (ref is feat/none/breaking_change_test)...
Skipping Git submodules setup
Executing "step_script" stage of the job script 00:00
$ git checkout -b $CI_COMMIT_BRANCH
Switched to a new branch 'feat/none/breaking_change_test'
$ nyx clean
time="2023-05-16T16:27:32-07:00" level=debug msg="selecting the command to run"
time="2023-05-16T16:27:32-07:00" level=debug msg="command to run is CLEAN"
time="2023-05-16T16:27:32-07:00" level=debug msg="Nyx.run(CLEAN)"
time="2023-05-16T16:27:32-07:00" level=debug msg="Nyx.clean()"
time="2023-05-16T16:27:32-07:00" level=debug msg="running command 'CLEAN'"
time="2023-05-16T16:27:32-07:00" level=debug msg="looking up command instance 'CLEAN' from cache"
time="2023-05-16T16:27:32-07:00" level=debug msg="no command instance 'CLEAN' found in cache"
time="2023-05-16T16:27:32-07:00" level=debug msg="creating new command instance 'CLEAN'"
time="2023-05-16T16:27:32-07:00" level=debug msg="instantiating the Git repository in '/home/gitlab-runner/builds/sU8YghRD/0/apps_engineering/ci-test'"
time="2023-05-16T16:27:32-07:00" level=debug msg="instantiating the initial state"
time="2023-05-16T16:27:32-07:00" level=debug msg="new Clean command object"
time="2023-05-16T16:27:32-07:00" level=debug msg="checking whether the Clean command is up to date"
time="2023-05-16T16:27:32-07:00" level=debug msg="command 'CLEAN' is up to date, skipping."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa1d634]
goroutine 1 [running]:
github.com/mooltiverse/nyx/modules/go/nyx/state.(*State).Summary(0xc0000a6300)
	/home/runner/work/nyx/nyx/modules/go/nyx/state/state.go:957 +0x494
main.main()
	/home/runner/work/nyx/nyx/modules/go/nyx/main/main.go:137 +0x449
Uploading artifacts for failed job 00:01
Uploading artifacts...
Runtime platform                                    arch=amd64 os=linux pid=4155952 revision=436955cb version=15.11.0
WARNING: build.env: no matching files. Ensure that the artifact path is relative to the working directory (/home/gitlab-runner/builds/xxx/ci-test) 
ERROR: No files to upload                          
Cleaning up project directory and file based variables 00:00
ERROR: Job failed: exit status 1
```

**Comment by flelli** [comment]
Hi @rrzal , this is a bug and I'll fix it soon.
Meanwhile, as a temporary workaround, you can avoid running the CLEAN task if no previous runs generated a state file or a version. That's because the original cause of this issue is that it dereferences the `version`, which is still `nil` at the time this CLEAN task runs.

**Comment by flelli** [comment]
Hi @rrzal , it's fixed in [2.3.5-hotfix217.1](https://github.com/mooltiverse/nyx/releases/tag/2.3.5-hotfix217.1) and GA in [2.3.5](https://github.com/mooltiverse/nyx/releases/tag/2.3.5) (still cooking as I'm writing)
